version: "3.8"

services:
  food-order-app-postgres:
    image: postgres:15.3-alpine
    container_name: food-order-app-postgres
    restart: on-failure
    environment:
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_DB: ${POSTGRES_DB}
    ports:
      - 5432:5432
    volumes:
      - ./backend/src/dataSource/queries/entrypoint.sql:/docker-entrypoint-initdb.d/db.sql
      - ./backend/data/postgres:/var/lib/postgresql/data
  food-order-app-postgres-pgadmin:
    image: dpage/pgadmin4:latest
    container_name: food-order-app-postgres-pgadmin
    restart: on-failure
    depends_on:
      - food-order-app-postgres
    environment:
      PGADMIN_DEFAULT_EMAIL: ${PGADMIN_DEFAULT_EMAIL}
      PGADMIN_DEFAULT_PASSWORD: ${PGADMIN_DEFAULT_PASSWORD}
    ports:
      - 8080:80
    volumes:
      - ./backend/data/pgadmin:/var/lib/pgadmin
  food-order-app-backend:
    build:
      context: ./backend
    container_name: food-order-app-backend
    restart: on-failure
    environment:
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_DB: ${POSTGRES_DB}
      DATABASE_PORT: ${DATABASE_PORT}
      DATABASE_HOST: ${DATABASE_HOST}
      NODE_ENV: production
    command: yarn start
    depends_on:
      - food-order-app-postgres
    ports:
      - 5001:5001
  food-order-app-dashboard:
    build:
      context: ./dashboard
    container_name: food-order-app-dashboard
    restart: on-failure
    command: yarn start
    ports:
      - 3000:3000
